entity_parser.go

```diff
@@ -365,20 +368,48 @@ func parseETLTag(tag string) (string, ETLState, error) {
 	return fullETLTag, etlState, nil
 }
 
+// parseTTLTag functions parses DOSA "ttl" tag
+func parseTTLTag(tag string) (string, time.Duration, error) {
+	fullTTLTag := ""
+	ttlTag := ""
+	matches := ttlPattern0.FindStringSubmatch(tag)
+	if len(matches) == 2 {
+		fullTTLTag = matches[0]
+		ttlTag = matches[1]
+	}
+
+	if len(matches) == 0 {
+		return "", NoTTL(), nil
+	}
+
+	// filter out "trailing comma"
+	ttlTag = strings.TrimRight(ttlTag, " ,")
+	ttl, err := time.ParseDuration(ttlTag)
+	if err != nil {
+		return "", NoTTL(), err
+	}
+
+	if ttl < 0 {
```

Comment:
should we return an error in this case? Why would we want to silently fail on a negative TTL? Is there a reasonable usecase for negative TTLs? Seems indicative of an error.